x-pack/plugins/secops/public/pages/hosts/index.tsx

```diff
@@ -3,66 +3,118 @@
+    <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
+      {({ events, kpiEventType, loading }) => (
+        <Pane1FlexContent data-test-subj="pane1FlexContent">
+          <VisualizationPlaceholder>
+            <HorizontalBarChart
+              loading={loading}
+              title="KPI event types"
+              width={490}
+              height={279}
+              barChartdata={
+                kpiEventType.map((i: KpiItem) => ({
+                  x: i.count,
+                  y: i.value,
+                })) as HorizontalBarChartData[]
+              }
+            />
+          </VisualizationPlaceholder>
+          <VisualizationPlaceholder>
+            <BasicTable
+              columns={getEventsColumns(dispatch)}
+              loading={loading}
+              pageOfItems={events}
+              sortField="host.hostname"
+              title="Events"
+            />
+          </VisualizationPlaceholder>
+          <Placeholders timelineId="pane2-timeline" count={8} myRoute="Hosts" />
+        </Pane1FlexContent>
+      )}
+    </EventsQuery>
+  ))
+);
 
-const eventsColumns: Columns[] = [
+const getEventsColumns = (dispatch: Dispatch) => [
   {
     name: 'Host name',
     sortable: true,
     truncateText: false,
     hideForMobile: false,
-    render: (item: EventItem) => (
-      <React.Fragment>{getOr('--', 'host.hostname', item)}</React.Fragment>
-    ),
+    render: (item: EventItem) => {
+      const hostName = getOr('--', 'host.hostname', item);
+      return (
+        <ProviderContainer
+          onClick={() => {
+            dispatch(
+              timelineActions.addProvider({
+                id: 'pane2-timeline',
+                provider: {
+                  enabled: true,
+                  id: `id-${hostName}`,
+                  name,
+                  negated: false,
+                  componentResultParam: 'events',
+                  componentQuery: EventsQuery,
+                  componentQueryProps: {
+                    sourceId: 'default',
+                    startDate,
+                    endDate,
+                    filterQuery: `{"bool":{"should":[{"match":{"host.name":"${hostName}"}}],"minimum_should_match":1}}`,
+                  },
+                  render: () => (
+                    <div data-test-subj="mockDataProvider">
+                      <EuiBadge color="primary">n/a</EuiBadge>
+                      <Text> {hostName} </Text>
+                    </div>
+                  ),
+                },
+              })
+            );
+          }}
+        >
+          {hostName}
+        </ProviderContainer>
+      );
+    },
   },
   {
     name: 'Event type',
     sortable: true,
     truncateText: true,
     hideForMobile: true,
-    render: (item: EventItem) => <React.Fragment>{getOr('--', 'event.type', item)}</React.Fragment>,
+    render: (item: EventItem) => (
+      <ProviderContainer>{getOr('--', 'event.type', item)}</ProviderContainer>
```

Comment:
Eventually we'll move the `--` into a constants file for a placeholder value. Still looking good!